src/lib.rs

```diff
@@ -138,12 +150,25 @@ impl PhaseVocoder {
 
         // push samples to input queue
         for chan in 0..input.len() {
+            // Zero pre-padding.
+            for _ in 0..self.frame_size {
+                self.in_buf[chan].push_back(0.0);
+                self.samples_waiting += 1;
+            }
+
             for samp in 0..input[chan].len() {
                 self.in_buf[chan].push_back(input[chan][samp].to_f64().unwrap());
                 self.samples_waiting += 1;
             }
+
+            // Zero post-padding
+            for _ in 0..self.frame_size {
+                self.in_buf[chan].push_back(0.0);
+                self.samples_waiting += 1;
+            }
         }
-        while self.samples_waiting >= 2 * self.frame_size * self.channels {
+
+        while self.samples_waiting > self.frame_size * self.channels {
```

Comment:
The previous loop condition was strange for me, so I've replaced it. Let me know if you have any issues with this.

Reply:
This seems strange to me too, but I'm getting panics from going out of array bounds with pvoc-plugins after this change. I think it's because of the strange implementation I wrote of overlap-add. We actually need slightly more than an entire frame when the windows don't line up with it perfectly.

Reply:
Ok, thanks for figuring this out. The panics you got should have been detected by the tests, so I'll add more tests (and fix it, of course).

Reply:
I've added a test that triggers the panic (when the `process` function uses the wrong while loop condition). I also did some debugging and it turns out that the panic was actually in the analysis step. 

In the synthesis step, `step_size * self.channels` input samples are popped, but this happens `self.time_res` times. After the second-to-last time, there still need to be `self.frame_size * self.channels` input samples for the last analysis step.